core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -63,6 +63,7 @@ class MapUnit {
     var attacksThisTurn = 0
     var promotions = UnitPromotions()
     var due: Boolean = true
+    var isAirUnitInCity: Boolean = false
```

Comment:
This property `isAirUnitInCity` is `false` for the ait units which were built in the city.
If you built the carrier in the city with air units, it will take them out since this property is `false`.

Reply:
The suggestion is:
1) to make it a general property (not specific to Air units) - that will require to move the assignment on [line 464](https://github.com/yairm210/Unciv/pull/1723/files/0936e421ca0fa5c4bd50b8789022617936d7fbb8#diff-886543b5b46160f44f1cc856e19216f5R464) out of `isAirUnit` condition
2) set it as `true` by default, since all new units appear in the cities. Also, in the name of compatibility: currently existing air units (in the save files) are in the cities only.

Reply:
Agreed and cherry-picked. Thanks for the catch!